tokio/src/io/blocking.rs

```diff
@@ -30,6 +30,7 @@ pub(crate) const DEFAULT_MAX_BUF_SIZE: usize = 2 * 1024 * 1024;
 enum State<T> {
     Idle(Option<Buf>),
     Busy(sys::Blocking<(io::Result<usize>, Buf, T)>),
+    Shutdown,
```

Comment:
Why is there a new state?

Reply:
At first, I changed the `Busy` state to:
```rust
Busy(Option<sys::Blocking<(io::Result<usize>, Buf, T)>>)
```
since `spawn_mandatory_blocking` optionally returns a `JoinHandle`. But now `Busy` is representing two states:
- `Busy(Some(jh))`: the task has been registered and will be picked up by one of the threads
- `Busy(None)`: the runtime is shutting down
I think it makes sense to only use `Busy` for the first one and introduce the `Shutdown` state to represent the second one.

Reply:
I guess under the current implementation, the `spawn_blocking` call panics and then you end up in the `Idle(None)` state.